ci/ci/build.py

```diff
@@ -89,10 +90,12 @@ def __init__(self, code, config_str, scope, requested_step_names=None):
         for step_config in config['steps']:
             step_params = StepParameters(code, scope, step_config, name_step)
             step = Step.from_json(step_params)
-            self.steps.append(step)
             name_step[step.name] = step
+            all_steps.append(step)
```

Comment:
This doesn't work either.  How are the runIfRequested steps excluded?  self.steps is no longer used, it's just rewritten below.